pyqg/tests/test_diagnostic_tools.py

```diff
@@ -21,3 +23,46 @@ def test_calc_ispec():
     spectrum_peak_idx = np.argmax(iso_spectrum)
     sinewave_freq_idx = np.argmin(np.abs(iso_wavenumbers - frequency))
     assert spectrum_peak_idx == sinewave_freq_idx
+
+def test_calc_ispec_units():
+    fixtures_path = f"{os.path.dirname(os.path.realpath(__file__))}/fixtures"
+
+    with open(f"{fixtures_path}/LayeredModel_params.pkl", 'rb') as f:
+        # Common set of parameters for each model
+        params = pickle.load(f)
+
+    m1 = pyqg.LayeredModel(nx=96, **params)
+    m2 = pyqg.LayeredModel(nx=64, **params)
+    m1.q = np.load(f"{fixtures_path}/LayeredModel_nx96_q.npy")
+    m2.q = np.load(f"{fixtures_path}/LayeredModel_nx64_q.npy")
+    for m in [m1, m2]:
+        m._invert()
+        m._calc_derived_fields()
+
+    for diagnostic in m1.diagnostics.keys():
+        if diagnostic == 'Dissspec':
+            continue
+
+        if m1.diagnostics[diagnostic]['dims'] == ('l','k'):
+            spec1 = m1.diagnostics[diagnostic]['function'](m1)
+            spec2 = m2.diagnostics[diagnostic]['function'](m2)
+            k1r, spec1r = calc_ispec(m1, spec1)
+            k2r, spec2r = calc_ispec(m2, spec2)
+
+            scales = [
+                np.abs(spec1).sum(),
+                np.abs(spec2).sum(),
+                np.trapz(np.abs(spec1r), k1r),
+                np.trapz(np.abs(spec2r), k2r)
+            ]
+
+            if max(scales) == 0:
+                continue
+
+            scale_ratio = max(scales) / min(scales)
+
+            assert scale_ratio < 10, \
+                f"calc_ispec should have correct units for {diagnostic}"
+            assert scale_ratio > 0.1, \
+                f"calc_ispec should have correct units for {diagnostic}"
```

Comment:
Let me know if this is the right assertion @dante831 -- I think I'm testing that the integral of the l,k spectra match the integral of the radial spectra, at least to one order of magnitude.